### ISSUE TITLE
[extension/observer/host] Can't be used inside container

### ISSUE THREAD

**Issue by sergeysedoy97** [open-telemetry/opentelemetry-collector-contrib#34058]
### Describe the issue you're reporting

[receiver/hostmetrics] contains special config field `root_path` and supports gopsutil envs like HOST_PROC

Is there any plan to add the container support for extension/observer/host?

**Comment by frzifus** [comment]
Maybe I am getting this wrong, but the hostmetrics receiver retrieves its data from the host. When using it inside a container, you may want to mount the hostfs as read-only into e.g. `/hostfs` and use this as `root_path`.

**Comment by sergeysedoy97** [comment]
> Maybe I am getting this wrong, but the hostmetrics receiver retrieves its data from the host. When using it inside a container, you may want to mount the hostfs as read-only into e.g. `/hostfs` and use this as `root_path`.

yes, receiver/hostmetrics can, but extension/observer/host **can't**

**Comment by atoulme** [comment]
I don't follow. Please clarify your request.
